sceptre/cli/list.py

```diff
@@ -80,52 +134,12 @@ def list_outputs(ctx, path, export):
     plan = SceptrePlan(context)
     responses = [response for response in plan.describe_outputs().values() if response]
 
-    # Legacy. This option was added in the initial commit of the project,
-    # although its intended use case is unclear. It may relate to a feature
-    # that had been removed prior to the initial commit.
     if export == "envvar":
-        for response in responses:
-            for stack in response.values():
-                for output in stack:
-                    write(
-                        "export SCEPTRE_{0}='{1}'".format(
-                            output.get("OutputKey"), output.get("OutputValue")
-                        ),
-                        "text",
-                    )
-
-    # Format outputs as !stack_output references.
+        process_export_envvar(responses)
```

Comment:
instead of 3 helper functions why not include the `if` conditions and create just 1 helper function?  i think the helper function provides more context with the `if` conditions.

Reply:
Yes, but that just moves the complexity problem somewhere else:

<img width="747" alt="Screenshot 2023-09-03 at 11 22 02 pm" src="https://github.com/Sceptre/sceptre/assets/2305330/4632c680-7a24-4f45-9ee0-5165bcc158d5">

Any other ideas?

Reply:
i think it's fine as is because i have no other ideas.  I'm wondering if @jfalkenstein might have something up his sleeve?

Reply:
I still prefer the first implementation, because I think it is easier to understand and more consistent with the rest of the functions in this CLI. I would have no problem with a `noqa` line. But I really don't mind.

@jfalkenstein, your thoughts?

Reply:
how about the idea of single helper function with `if`+`for`s+`noqa`?

Reply:
So here's my two-cents on what _I'd_ do, based mostly on the Single Responsibility Principle:
1. I'd make a generator called `iterate_stack_outputs` that iterates tuples of `(stack_name, output_dict)`. This basically flattens out the loops within loops within loops you keep repeating.
2. I'd make three functions: `write_envvar(output_dict)`, `write_stack_output_resolver(stack_name, output_dict)`, and `write_external_stack_output_resolver(stack_name, output_dict)`

Thus, you're separating the concerns of (1) looping and (2) writing, one for each output. 

The generator in point 1 above would look something like this:

```python
def iterate_stack_outputs(responses: list[dict[str, list[dict]]]):
    for response in responses:
        for stack_name, values in response.items()
            for value in values:
                yield stack_name, value
```

This way you have a single iterator that flattens it all out.

Reply:
@jfalkenstein , @zaro0508 Updated as per Jon's suggestions. Thoughts? I can live with any of these solutions, although I think I am still in favour of the original for the same reasons. I think we've made the code a bit harder to understand in this version despite that it is true that we have refactored out some duplication using the iterator.

Reply:
I wouldn't mind this one so much if it would let me use nested functions but sadly it still fails the complexity check if I move these to be nested functions.

Reply:
I'm not a fan of @jfalkenstein suggestion.  I would definitely have a harder time understanding it vs the original simple implementation.  I would prefer to go with the initial simpler implementation.  With that being said i'm also ok with  @jfalkenstein's suggestion as long as we add some comments explaining that we implemented it that way to    avoiding `noqa` failures.